src/main/java/io/zksync/transport/receipt/ZkSyncPollingTransactionReceiptProcessor.java

```diff
@@ -0,0 +1,79 @@
+package io.zksync.transport.receipt;
+
+import java.util.concurrent.CompletableFuture;
+
+import io.zksync.domain.transaction.TransactionDetails;
+import io.zksync.exception.ZkSyncException;
+import io.zksync.provider.AsyncProvider;
+import io.zksync.transport.ZkTransactionStatus;
+import io.zksync.wallet.ZkASyncWallet;
+
+public class ZkSyncPollingTransactionReceiptProcessor extends ZkSyncTransactionReceiptProcessor {
+
+    protected final long sleepDuration;
+    protected final int attempts;
+
+    public ZkSyncPollingTransactionReceiptProcessor(AsyncProvider provider, long sleepDuration, int attempts) {
+        super(provider);
+
+        this.sleepDuration = sleepDuration;
+        this.attempts = attempts;
+    }
+
+    public ZkSyncPollingTransactionReceiptProcessor(ZkASyncWallet wallet, long sleepDuration, int attempts) {
+        this(wallet.getProvider(), sleepDuration, attempts);
+    }
+
+    public ZkSyncPollingTransactionReceiptProcessor(AsyncProvider provider) {
+        this(provider, 100, Integer.MAX_VALUE);
+    }
+
+    public ZkSyncPollingTransactionReceiptProcessor(ZkASyncWallet wallet) {
+        this(wallet.getProvider());
+    }
+
+    @Override
+    public CompletableFuture<TransactionDetails> waitForTransaction(String hash, ZkTransactionStatus status) {
+        final CompletableFuture<TransactionDetails> result = CompletableFuture.supplyAsync(() -> {
+            TransactionDetails details = this.getTransactionDetails(hash).join();
+            for (int i = 0; i < attempts; i++) {
+                if (!details.getExecuted()) {
```

Comment:
How will it behave on failed transactions?

Reply:
It will return transaction details on committed or verified with actual success status. This processor works like the similar one in Web3J